qiskit_experiments/data_processing/processor_library.py

```diff
@@ -12,36 +12,101 @@
 
 """A collection of functions that return various data processors."""
 
-from qiskit.qobj.utils import MeasLevel
+from qiskit.qobj.utils import MeasLevel, MeasReturnType
 
+from qiskit_experiments.framework import ExperimentData, Options
 from qiskit_experiments.data_processing.exceptions import DataProcessorError
 from qiskit_experiments.data_processing.data_processor import DataProcessor
 from qiskit_experiments.data_processing import nodes
 
 
 def get_processor(
-    meas_level: MeasLevel = MeasLevel.CLASSIFIED,
-    meas_return: str = "avg",
-    normalize: bool = True,
+    experiment_data: ExperimentData, analysis_options: Options, index: int = -1
 ) -> DataProcessor:
     """Get a DataProcessor that produces a continuous signal given the options.
 
     Args:
-        meas_level: The measurement level of the data to process.
-        meas_return: The measurement return (single or avg) of the data to process.
-        normalize: Add a data normalization node to the Kerneled data processor.
+        experiment_data: The experiment data that holds all the data and metadata needed
+             to determine the data processor to use to process the data for analysis.
+        analysis_options: The analysis options with which to analyze the data. The options that
+             are relevant for the configuration of a data processor are:
+             - normalization (bool): A boolean to specify if the data should be normalized to
+               the interval [0, 1]. The default is True. This option is only relevant if
+               kerneled data is used.
+             - outcome (string): The measurement outcome that will be passed to a Probability node.
+             - restless_threshold (float): If (repetition delay / T1) is below a given threshold for
+               all physical qubit T1 values the experiment is recognized as a restless experiment and
+               will be analyzed accordingly.
+        index: The index of the job for which to get a data processor.
 
     Returns:
-        An instance of DataProcessor capable of dealing with the given options.
+        An instance of DataProcessor capable of processing the data for the corresponding job.
+
+    Notes:
+        The following relevant arguments are extracted from the experiment_data metadata run options:
+            - meas_level (MeasLevel): The measurement level of the data to process.
+            - meas_return (MeasReturnType): The measurement return (single or avg) of the data to process.
+            - init_qubits (bool): If False, the qubits are not reset to the ground state after a measurement.
+            - physical qubits: The physical qubits used in the experiment.
+            - memory (bool): If True, single-shot measurement bitstrings are returned.
+            - rep_delay (float): The delay between a measurement and the subsequent circuit.
```

Comment:
Nice. Can you also mention the default values that the code assumes?

Reply:
Yes, see dac22d9.